### ISSUE TITLE
[Bug]: Performance is quite severely degraded with lots of `contentFile` entries

### ISSUE THREAD

**Issue by atrauzzi** [NuGet/Home#11543]
### NuGet Product Used

dotnet.exe

### Product Version

6.0.100

### Worked before?

N/A

### Impact

It's more difficult to complete my work - although one might argue it's basically unusable.

### Repro Steps & Context

Take a look at the following nuget package: https://github.com/atrauzzi/protoculture-dotnet-postgres/suites/5130643290/artifacts/154430129

Out of necessity, it contains *a lot* of `contentFiles` entries as it bundles in native, portable builds of postgres.

Unfortunately when I add it to a project, it seems like `dotnet restore` now grints to a halt for quite some time, slowing the dev loop substantially and presenting a usability concern.

![image](https://user-images.githubusercontent.com/115199/151912937-3a1dd43d-63f1-4b0d-a7f5-58a7fbf1c16a.png)

Barring any technique I'm missing out on that I can leverage in [my `.csproj` file](https://github.com/atrauzzi/protoculture-dotnet-postgres/blob/main/Protoculture.Postgres.Embedded/Protoculture.Postgres.Embedded.csproj#L27-L35), it seems like the nuget client could use some TLC when it comes to `contentFiles`.

### Verbose Logs

N/A

**Comment by atrauzzi** [comment]
The package is now up on nuget.org if you wish to work with something "nearby": https://www.nuget.org/packages/Protoculture.Postgres.Embedded/0.0.1-pre

Restore speed is still very bad.

**Comment by nkolev92** [comment]
I can see repro the slow restore. 

I see 4348 different files and 4348 different lines in the nuspec, definitely on the higher end I've dealt with :D 

Took about 5 mins on my machine.  Force restore also runs for a while, so it's probably somewhere in the selection code.

**Comment by atrauzzi** [comment]
Yeah, it's definitely a lot, and even after trimming things down.  But I don't know why I should be limited. Maybe there needs to be support for globs at this level (and a way to propagate them from `.csproj` files)?

Would runtimes be subject to any similar issues?

**Comment by nkolev92** [comment]
The slow task is the compatibility checking.

I don't know if the runtimes has a similar problem, as I haven't gone in depth, but it should be running similar codepaths, so if I had to guess, likely similar problems.

**Comment by atrauzzi** [comment]
Would love to have that included for consideration here as I received advice from @damianedwards in a small Twitter thread about switching to using it... 🙂🙏

**Comment by DamianEdwards** [comment]
If the runtime assets are split into their own packages (one per runtime id) I'd hope the restore cost would be reduced. That said, it does seem like there might be an opportunity to identify and address the underlying cause in NuGet.

**Comment by atrauzzi** [comment]
I originally encountered this issue when only one set of postgres binaries were being included.  I'd say the issue is present for the most trivial of - at least my - scenarios.

**Comment by nkolev92** [comment]
Another report in https://github.com/NuGet/Home/issues/12083.
